src/deadline_test_fixtures/deadline/resources.py

```diff
@@ -1167,6 +1231,61 @@ def assert_log_contains(
             else:
                 return
 
+    def assert_log_does_not_contain(
+        self,
+        *,
+        logs_client: BaseClient,
+        expected_pattern: re.Pattern | str,
+        assert_fail_msg: str = "Expected message found in session log",
+        consistency_wait_time: timedelta = timedelta(seconds=3),
+    ) -> None:
+        """
+        Asserts that the expected regular expression pattern does not exist in the job's session log.
+
+        This method accounts for the eventual-consistency of CloudWatch log delivery and availability
+        through CloudWatch APIs by a configurable wait time. The method does an initial immediate
+        check, then waits for the configured consistency wait time before checking again, if the wait
+        time is greater than zero. If neither check (or one check if wait time is zero) matches the
+        expected pattern then the log is assumed to not contain the given line.
+
+        Args:
+            logs_client (botocore.clients.BaseClient): CloudWatch logs boto client
+            expected_pattern (re.Pattern | str): Either a regular expression pattern string, or a
+                pre-compiled regular expression Pattern object. This is pattern is searched against
+                each of the job's session logs, contatenated as a multi-line string joined by
+                a single newline character (\\n).
+            assert_fail_msg (str): The assertion message to raise if the pattern is found
+                The CloudWatch log group name is appended to the end of this message to assist
+                with diagnosis. The default is "Expected message found in session log".
+            consistency_wait_time (datetime.timedelta): Wait time between first and second check.
+                Default is 3s, wait times opperates in second increments.
```

Comment:
To remain consistent with `assert_single_task_log_contains` https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/65b93eab837387e2776e427d273dbbeba8d73ef7/src/deadline_test_fixtures/deadline/resources.py#L1166 I used [timedelta.total_seconds()](https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds) which has a 1 second resolution. 3 seconds was chosen to appear visually consistent with `assert_single_task_log_contains` whose default is 300 milliseconds

Reply:
Ahh I see what you're getting at now. Though `total_seconds` actually explicitly supports greater degrees of resolution since it's equivalent to `td / timedelta(seconds=1)`, specifically they're not flooring it:

```
>>> import datetime
>>> datetime.timedelta(microseconds=1).total_seconds()
1e-06
>>> datetime.timedelta(milliseconds=1).total_seconds()
0.001
```

From some quick math, the total time for the positive assertion takes 4.5s (4 tries: 0.3s + 0.6s + 1.2s + 2.4s) before it gives up. Should we just match that for the negative case? Personally I'd love more data on how long the positive assertion generally takes, but it seems like a safe default until we have that data.

Reply:
Sure that is fair I can match that and put 4.5 as the default